plugin/service/storage/plugin.go

```diff
@@ -41,6 +42,51 @@ type StoragePlugin struct {
 	pb.UnimplementedStoragePluginServiceServer
 }
 
+// NormalizeStorageBucketData is called by boundary before calling OnCreate or OnUpdate
+//
+// minio supports 3 attributes:
+//   - endpoint_url
+//   - region
+//   - disable_credential_rotation
+//
+// of these, only endpoint is required and normalized. region and disable_credential_rotation
+// are passed as-is, if they were provided
+func (sp *StoragePlugin) NormalizeStorageBucketData(ctx context.Context, req *pb.NormalizeStorageBucketDataRequest) (*pb.NormalizeStorageBucketDataResponse, error) {
+	sa, err := getStorageAttributes(req.GetAttributes())
+	if err != nil {
+		return nil, err
+	}
+
+	// re-add the scheme since getStorageAttributes removes it
+	if sa.UseSSL {
+		sa.EndpointUrl = "https://" + sa.EndpointUrl
+	} else {
+		sa.EndpointUrl = "http://" + sa.EndpointUrl
+	}
+
+	if sa.EndpointUrl, err = parseutil.NormalizeAddr(sa.EndpointUrl); err != nil {
+		return nil, status.New(codes.InvalidArgument, fmt.Sprintf("failed to normalize provided %s: %s", ConstEndpointUrl, err.Error())).Err()
+	}
+
+	fields := map[string]*structpb.Value{
```

Comment:
I might be missing something here, but why not just use the input and change the endpoint url within and return the rest unchanged?

This avoids the code below, which also means if extra fields get added down the line, they would get eaten here.

Reply:
It seems like a good idea to just use this to normalize attributes we know need to be normalized and set that. Then have the create or update path do the explicit check for each other attribute.